internal/system/agent/init.go

```diff
@@ -32,9 +32,15 @@ var Configuration *ConfigurationStruct
 var LoggingClient logger.LoggingClient
 var Manifest *ManifestStruct
 var Conf = &ConfigurationStruct{}
-var nc general.GeneralClient
 var ec interfaces.ExecutorClient
-var mcc general.GeneralClient
+var gccc general.GeneralClient
+var gccd general.GeneralClient
+var gccm general.GeneralClient
+var gcec general.GeneralClient
+var gced general.GeneralClient
+var gcsl general.GeneralClient
+var gcsn general.GeneralClient
+var gcss general.GeneralClient
```

Comment:
For this first pass, the above is fine. Going forward, consider using a map[string]general.GeneralClient. Set the key to one of the service key constants and then in your service.go loops for metrics and config, use the key to fetch the client from the map and make the relevant call. This will be easier to extend if we add more services.

The initializeClients function below could use a similar strategy to populate the map.

Reply:
Completely agree with @tsconn23!